src/backend/engine/index.js

```diff
@@ -1,3 +1,23 @@
-define(["./TestableFile"], function(Testable) {
-    alert("calculation result is " + new Testable(function() { return 3; }).activity());
+define(["mod_parser", "mod_executor", "mod_data_structures"], function(parser, executor, dataStructures) {
+    
+    var process;
+
+    function runProgram(source) {
```

Comment:
I think the name is confusing. runProgram should be a (maybe frontend side) function which every specified interval call .nextStep(), and this one should be named compileProgram. We want to do this at the moment when user click Compile button.